File: core/test/org.eclipse.vjet.test.core.jstojava/src/org/eclipse/vjet/dsf/jst/validation/vjo/samples/vjoPro/samples/dom/HTMLandDomEx6.java
/*******************************************************************************
 * Copyright (c) 2005-2012 eBay Inc.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 *******************************************************************************/

package org.eclipse.vjet.dsf.jst.validation.vjo.samples.vjoPro.samples.dom;




import java.util.List;

import org.eclipse.vjet.dsf.jsgen.shared.ids.MethodProbIds;
import org.eclipse.vjet.dsf.jsgen.shared.validation.vjo.VjoSemanticProblem;
import org.eclipse.vjet.dsf.jst.validation.vjo.VjoValidationBaseTester;
import org.junit.Before;
import org.junit.Test;




/**
 * HTMLandDomEx6.java
 * 
 * @author <a href="mailto:liama@ebay.com">liama</a>
 * @since JDK 1.5
 */
//@Category( { P3, FAST, UNIT })
//@ModuleInfo(value="DsfPrebuild",subModuleId="JsToJava")
public class HTMLandDomEx6 extends VjoValidationBaseTester {

    @Before
    public void setUp() {
        expectProblems.clear();
        expectProblems.add(createNewProblem(MethodProbIds.UndefinedMethod, 13,
                0));
        expectProblems.add(createNewProblem(MethodProbIds.UndefinedMethod, 30,
                0));
        expectProblems.add(createNewProblem(MethodProbIds.UndefinedMethod, 29,
                0));
        expectProblems.add(createNewProblem(MethodProbIds.UndefinedMethod, 14,
                0));
        expectProblems.add(createNewProblem(MethodProbIds.UndefinedMethod, 28,
                0));
    }

    @Test
    //@Category( { P3, FAST, UNIT })
    //@Description("Test VJO Sample project, To validate false positive ")
    public void testHTMLandDomEx6() {
        List<VjoSemanticProblem> problems = getVjoSemanticProblem(
                VjoValidationBaseTester.VJLIB_FOLDER, "vjoPro.samples.dom.",
                "HTMLandDomEx6.js", this.getClass());
        assertProblemEquals(expectProblems, problems);
    }
}

File: core/test/org.eclipse.vjet.test.core.jstojava/vjLibTestFiles/vjoPro/samples/dom/HTMLandDomEx6.js
/*******************************************************************************
 * Copyright (c) 2012 eBay Inc. and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     eBay Inc. - initial API and implementation
 *******************************************************************************/
vjo.ctype('vjoPro.samples.dom.HTMLandDomEx6') //< public
.needs(['vjoPro.dsf.utils.URL','vjoPro.dsf.document.Element'])
.props({
/**
* @return void
* @access public
* @param {String} psUrlId
* @param {String} psArgNameId
*
*/
//> public void readArgument(String psUrlId,String psArgNameId)
readArgument : function(psUrlId, psArgNameId) {
var sUrl = vjoPro.dsf.document.Element.get(psUrlId).value;
var sArgName = vjoPro.dsf.document.Element.get(psArgNameId).value;
alert(vjoPro.dsf.utils.URL.getArg(sUrl, sArgName));
},

/**
* @return void
* @access public
* @param {String} psUrlId
* @param {String} psArgNameId
* @param {String} psArgValId
*
*/
//> public void addArgument(String psUrlId,String psArgNameId,String psArgValId)
addArgument : function(psUrlId, psArgNameId, psArgValId) {
var E = vjoPro.dsf.document.Element.get(psUrlId);
var sArgName = vjoPro.dsf.document.Element.get(psArgNameId).value;
var sArgVal = vjoPro.dsf.document.Element.get(psArgValId).value;
E.value = vjoPro.dsf.utils.URL.addArg(E.value, sArgName, sArgVal);
}

})
.endType();
